### DISCUSSION TITLE
Don't use object key/values for `conditions` when using `defineProperties` from "@vanilla-extract/sprinkles".

### DISCUSSION THREAD

**Discussion by alex-mcgovern** [vanilla-extract-css/vanilla-extract#1109] [Ideas]
Use of an object implies that ordering is not important. Was recently stung by this when introducing sorting of object keys into a component library codebase, which broke responsive sprinkles conditions. If ordering is important here, I would expect something like an array of tuples or a map.

Is just a suggestion, but I think the mental model would be more clearly understood. :)

**Comment by graup** [comment]
Interesting point. I totally can see your point of view but I'm not sure everyone would agree with the statement "an object implies that ordering is not important." The [spec](https://stackoverflow.com/a/38218582/700283) certainly doesn't. I think automatic "object key sorting" is highly debatable and I certainly would not add it to my codebases.

That said, allowing Maps in place of objects in those APIs wouldn't hurt.

**Reply by alex-mcgovern** [reply]
This is a fair response, and valid point on the spec.
